### DISCUSSION TITLE
Configuring With AWS SES

### DISCUSSION THREAD

**Discussion by ajinbmc** [postalserver/postal#2402] [Help with using Postal]
Hi Team,

I am trying to integrate AWS SES with Postal, Is There any way to do this?
I'm also receiving this error while trying to send a test email 
No SMTP servers were available for gmail.com. Tried gmail-smtp-in.l.google.com, alt1.gmail-smtp-in.l.google.com, alt2.gmail-smtp-in.l.google.com, alt3.gmail-smtp-in.l.google.com, and alt4.gmail-smtp-in.l.google.com

can you guys please help me with the issue

Thank you!

**Comment by willpower232** [comment]
> No SMTP servers were available for gmail.com

Either google is blocking your server provider or your server provider is preventing you from sending messages.

SES and Postal do very similar jobs, its unlikely you would get much out of using them both in the same process. I don't think SES would like you create a password-less relay that Postal expects to use.

**Reply by arthurzenika** [reply]
I think there is a lot of user-centric features in Postal that are not present in SES, so I think they can be complimentary. For the password-less relay, you are right, SES will only expose an authenticated SMTP server. If postal doesn't support authenticated outgoing SMTP relays, a hack would be to use a locally installed nullmailer https://packages.debian.org/source/stable/nullmailer where the SES credentials are configured (see debian's wiki https://wiki.debian.org/nullmailer with the gmail authenticated example).

**Comment by ajinbmc** [comment]
Hey Will,

Thank you for your response, I was able to configure SES to postal , but the error still persist.
I'm receiving this error 
Creating postal_runner_run ... done
usage: postal test-app-smtp [email address]
ERROR: 1
whenever i use the command postal test-app-smtp from terminal. Can you please help me with this

**Reply by willpower232** [reply]
If the usage text is showing then you need to make sure you have called the command correctly, in this case specifying an email address to send a testing email to.

**Comment by imvikysharma** [comment]
To integrate AWS SES with Postal, you can set up SES as an SMTP endpoint in Postal. Here are the general steps to follow:

Create a new SMTP endpoint in Postal and choose "Custom SMTP" as the provider.
Then Configure the endpoint to use the SES SMTP server and port (smtp.gmail.com:587 or smtp.gmail.com:465 for TLS/SSL).
Then Configure the endpoint to use your SES SMTP credentials (SMTP username and password).
Update your Postal delivery settings to use the new SES endpoint.

I would like to recommend a book on AWS

https://www.amazon.com/dp/B0BVJKH8PY?maas=maas_adg_AA06FAB7196A55FE3DBBEE7958CED70A_afap_abs&ref_=aa_maas&tag=maas

**Reply by ajinbmc** [reply]
Hi Viky , Thank you for the response, Can you please give me an idea on how to configure a Custom SMTP in postal . i'm not seeing any option for that , I've added these lines in my postal.yml file 
smtp:
  # Configure SES as the SMTP server
  host: email-smtp.us-east-2.amazonaws.com
  port: 587
  username: xyz
  password: xyz
  authentication: LOGIN
  enable_starttls_auto: true
  from_name: Postal
  from_address: test@postal.domain.com
  smtp_use_tls: true

**Reply by imvikysharma** [reply]
Make sure to save your changes to the "postal.yml" file and restart the Postal service to apply the new SMTP settings.

**Reply by arthurzenika** [reply]
How do you manage Bounce management ? I would presume all outgoing emails are seen as sent since SES can't tell when receiving the email from postal if it can be sent or not. 

I can see in Mautic that they use Amazon's webhooks to handle that scenario https://docs.mautic.org/en/channels/emails/bounce-management#amazon-webhook ... any chance of having something similar in postal ?
